stores/redis/redis.go

```diff
@@ -0,0 +1,344 @@
+	}
+	entryKey := entryPathPrefix + id
+	logrus.Debugf("Adding key '%s': %s", entryKey, raw)
+	err = r.createValue(entryKey, raw)
+	if err != nil {
+		msg := fmt.Sprintf("Failed to set key '%s' for user '%s': %v", entryKey, userIdentifier, err)
+		logrus.Error(msg)
+		return errors.Wrap(err, msg)
+	}
+
+	// add path->user mapping
+	userKey := entryUserPrefix + id
+	logrus.Debugf("Adding key '%s': %s", userKey, raw)
+	err = r.createValue(userKey, []byte(userIdentifier))
+	if err != nil {
+		msg := fmt.Sprintf("Failed to set key '%s' for user '%s': %v", userKey, userIdentifier, err)
+		logrus.Error(msg)
+		return errors.Wrap(err, msg)
+	}
+
+	// add the entry to the SET of entries for the useridentifier
+	userEntriesKey := userToEntriesPrefix + userIdentifier
+	logrus.Debugf("Adding entry '%s' to set of entries for user '%s'", id, userIdentifier)
+	result := r.c.SAdd(userEntriesKey, id)
+	if result.Err() != nil {
+		msg := fmt.Sprintf("Failed to add entry '%s' for user '%s': %v", id, userIdentifier, result.Err())
+		logrus.Error(msg)
+		return errors.Wrap(result.Err(), msg)
+	}
+	logrus.Debugf("Successfully added entry '%s' to set '%s'", id, userEntriesKey)
+	return nil
+}
+
+func (r *RedisStore) DeleteEntry(id string) error {
+	// delete the id-to-url mapping
+	entryKey := entryPathPrefix + id
+	err := r.delValue(entryKey)
+	if err != nil {
+		msg := fmt.Sprintf("Could not delete entry id %s: %v", id, err)
+		logrus.Error(msg)
+		return errors.Wrap(err, msg)
+	}
+	// delete the visitors list for the id
+	entryVisitsKey := entryVisitsPrefix + id
+	err = r.delValue(entryVisitsKey)
+	if err != nil {
+		msg := fmt.Sprintf("Could not delete visitors list for id %s: %v", id, err)
+		logrus.Error(msg)
+		return errors.Wrap(err, msg)
+	}
+
+	// get the user for the id
+	userKey := entryUserPrefix + id
+	var userIdentifier string
+	userIdentifier, err = r.c.Get(userKey).Result()
+	if err != nil {
+		msg := fmt.Sprintf("Could not fetch id to user mapping for id '%s': %v", id, err)
+		logrus.Error(msg)
+		return errors.Wrap(err, msg)
+	}
+
+	// delete the entry from set of entries for the user
+	userEntriesKey := userToEntriesPrefix + userIdentifier
+	err = r.c.SRem(userEntriesKey, id).Err()
+	if err != nil {
+		msg := fmt.Sprintf("Could not remove entry '%s' from list of entries for user '%s': %v", id, userIdentifier, err)
+		logrus.Error(msg)
+		return errors.Wrap(err, msg)
+	}
+
+	// delete the id-to-user mapping
+	err = r.delValue(userKey)
+	if err != nil {
+		msg := fmt.Sprintf("Could not delete the path-to-user mapping for entry '%s': %v", id, err)
+		logrus.Error(msg)
+		return errors.Wrap(err, msg)
+	}
+
+	return err
+}
+
+func (r *RedisStore) GetEntryByID(id string) (*shared.Entry, error) {
+	var entry *shared.Entry
+	var visitor *shared.Visitor
+	var lastVisit time.Time
+	var raw []byte
+	var err error
+
+	entryKey := entryPathPrefix + id
+	logrus.Debugf("Fetching key: '%s'", entryKey)
+	result := r.c.Get(entryKey)
+	raw, err = result.Bytes()
```

Comment:
I would simplify the Variable declarations to `:=` and remove the declaration 5 lines above.

Reply:
done